jwt/validate.go

```diff
@@ -184,5 +187,51 @@ func Validate(t Token, options ...ValidateOption) error {
 		}
 	}
 
+	for _, v := range claimsValidators {
+		if err := v.Validate(t); err != nil {
+			return err
+		}
+	}
+
 	return nil
 }
+
+// ClaimsValidator describes interface to validate Token.
+type ClaimsValidator interface {
+	Validate(Token) error
+}
+
+// SingleClaimValidator describes interface to validate a claim in Token.
+type SingleClaimValidator interface {
+	Validate(interface{}) error
+}
+
+// SingleClaimValidatorFunc is a named type to wrap raw function as SingleClaimValidator.
+type SingleClaimValidatorFunc func(interface{}) error
+
+func (f SingleClaimValidatorFunc) Validate(c interface{}) error {
+	return f(c)
+}
```

Comment:
SingleStringClaimValidator, SingleFloat64ClaimValidator, it would be convenient to have those.